server/api/api/endpoints/logs.py

```diff
@@ -79,6 +79,10 @@ async def get_log(
         server.api.api.deps.get_db_session
     ),
 ):
+    if offset < 0:
```

Comment:
Consider using:
```
offset: int = Query(0, ge=0),
```
Which will enforce `>=0` already in FastAPI, and should also display the limitation in the Swagger page. Only thing is to validate the error message is clear in that case.